controllers/drplacementcontrol_controller.go

```diff
@@ -611,13 +611,18 @@ func (r *DRPlacementControlReconciler) Reconcile(ctx context.Context, req ctrl.R
 
 	ensureDRPCConditionsInited(&drpc.Status.Conditions, drpc.Generation, "Initialization")
 
-	placementObj, err := r.ownPlacementOrPlacementRule(ctx, drpc, logger)
+	placementObj, err := getPlacementOrPlacementRule(ctx, r.Client, drpc, logger)
 	if err != nil && !(errors.IsNotFound(err) && !drpc.GetDeletionTimestamp().IsZero()) {
 		r.recordFailure(drpc, placementObj, "Error", err.Error(), nil, logger)
 
 		return ctrl.Result{}, err
 	}
 
+	err = r.setDRPCOwnerAndAnnotatePlacement(ctx, drpc, placementObj, logger)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
 	if isBeingDeleted(drpc, placementObj) {
```

Comment:
Can we simplify this check?

If the drpc will be deleted automatically, we don't care if the placement was deleted when reconcile, since the system ensures that the drpc is deleted in this case. So we can simplify the code to check if the drpc is deleted here:

    if drpc.IsDeleted() {
        ...
    }

Reply:
Well, the UI has a check mark to also delete the Placement.  If that is not checked, the placement will not be deleted.

Reply:
We also potentially need to look at backward compatibility, to ensure we need not worry about Placement deletion anymore.